packages/workflow/src/records/fhir.ts

```diff
@@ -76,6 +94,112 @@ function getFHIRValueField(value: unknown) {
   throw new Error('Invalid value type')
 }
 
+export async function mergeChangedResourcesIntoRecord(
+  record: ValidRecord,
+  unsavedChangedResources: Bundle,
+  practitionerResourcesBundle: SavedBundle<SavedLocation | SavedPractitioner>
+) {
+  const responseBundle = await sendBundleToHearth(unsavedChangedResources)
+
+  const changedResources = await toSavedBundle(
+    unsavedChangedResources,
+    responseBundle
+  )
+
+  const recordWithChangedResources = mergeBundles(record, changedResources)
+
+  return mergeBundles(recordWithChangedResources, practitionerResourcesBundle)
+}
+
+export async function withPractitionerDetails<T extends Task>(
+  task: T,
+  token: string
+): Promise<[T, SavedBundle<SavedLocation | SavedPractitioner>]> {
+  const userOrSystem = await getUserOrSystem(token)
+  const newTask: T = {
+    ...task,
+    identifier: task.identifier.filter(
+      ({ system }) =>
+        system !== 'http://opencrvs.org/specs/id/system_identifier'
+    )
+  }
+  if (isSystem(userOrSystem)) {
+    const { name, username, type } = userOrSystem
+    newTask.identifier.push({
+      system: 'http://opencrvs.org/specs/id/system_identifier',
+      value: JSON.stringify({
+        name,
+        username,
+        type
+      })
+    })
+    return [
+      newTask,
+      {
+        type: 'document',
+        resourceType: 'Bundle',
+        entry: []
+      }
+    ]
+  }
+  const user = userOrSystem
+  const practitioner = (await getLoggedInPractitionerResource(
+    token
+  )) as SavedPractitioner
+  const relatedLocations = (await getPractitionerLocations(
+    user.practitionerId
+  )) as [SavedLocation]
+  const office = relatedLocations.find((l) =>
+    l.type?.coding?.some(({ code }) => code === 'CRVS_OFFICE')
+  )
+  if (!office) {
+    throw internal('Office not found for the requesting user')
+  }
+  const officeLocationId = office.partOf?.reference.split('/').at(1)
+  const officeLocation = relatedLocations.find((l) => l.id === officeLocationId)
+  if (!officeLocation) {
+    throw internal(
+      'Parent location of office not found for the requesting user'
+    )
+  }
+  newTask.extension.push(
+    ...([
+      {
+        url: 'http://opencrvs.org/specs/extension/regLastUser',
+        valueReference: {
+          reference: `Practitioner/${user.practitionerId as UUID}`
+        }
+      },
+      {
+        url: 'http://opencrvs.org/specs/extension/regLastLocation',
+        valueReference: {
+          reference: `Location/${resourceIdentifierToUUID(
+            office.partOf!.reference
+          )}`
+        }
+      },
+      {
+        url: 'http://opencrvs.org/specs/extension/regLastOffice',
+        //@todo: make this field required
```

Comment:
Is this `@TODO` somewhere in a future PR or in this one

Reply:
hm not sure if I have addressed this in a PR or not